assemble/bin/accumulo-cluster

```diff
@@ -27,598 +35,517 @@ Options:
+  accumulo-cluster start --sservers=group1                 # start all group1 sservers
+  accumulo-cluster start --local --manager --tservers      # Start the local manager and local tservers
+
 EOF
 }
 
+function parse_args {
+  DEBUG=0
+  ARG_LOCAL=0
+  ARG_ALL=0
+  ARG_MANAGER=0
+  ARG_GC=0
+  ARG_MONITOR=0
+  ARG_COORDINATOR=0
+  ARG_TSERVER=0
+  ARG_TSERVER_GROUP=""
+  ARG_SSERVER=0
+  ARG_SSERVER_GROUP=""
+  ARG_COMPACTOR=0
+  ARG_COMPACTOR_GROUP=""
+
+  PARSE_OUTPUT=$(getopt -o "" --long "dry-run,all,local,manager,gc,monitor,compaction-coordinator,no-tservers,tservers,sservers::,compactors::" -n 'accumulo-cluster' -- "$@")
+  eval set -- "$PARSE_OUTPUT"
+
+  while true; do
+    case "$1" in
+      --dry-run)
+        DEBUG=1
+        debug "args: $PARSE_OUTPUT"
+        shift 1
+        ;;
+      --all)
+        ARG_ALL=1
+        shift 1
+        ;;
+      --local)
+        ARG_LOCAL=1
+        shift 1
+        ;;
+      --manager)
+        ARG_MANAGER=1
+        shift 1
+        ;;
+      --gc)
+        ARG_GC=1
+        shift 1
+        ;;
+      --monitor)
+        ARG_MONITOR=1
+        shift 1
+        ;;
+      --compaction-coordinator)
+        ARG_COORDINATOR=1
+        shift 1
+        ;;
+      --no-tservers)
+        echo -- "--no-tservers is deprecated. Please specify the servers you wish to manage instead"
+        ARG_MANAGER=1
+        ARG_GC=1
+        ARG_MONITOR=1
+        ARG_COORDINATOR=1
+        ARG_SSERVER=1
+        ARG_COMPACTOR=1
+        shift 1
+        ;;
+      --tservers)
+        ARG_TSERVER=1
+        shift 1
+        ;;
+      --sservers)
+        ARG_SSERVER=1
+        if [[ -n $2 ]]; then
+          ARG_SSERVER_GROUP=$2
+        fi
+        shift 2
+        ;;
+      --compactors)
+        ARG_COMPACTOR=1
+        if [[ -n $2 ]]; then
+          ARG_COMPACTOR_GROUP=$2
+        fi
+        shift 2
+        ;;
+      --)
+        shift
+        break
+        ;;
+      *)
+        echo "Unhandled option: $1"
+        print_usage
+        exit 1
+        ;;
+    esac
+  done
+
+  # All and any of the others are mutually exclusive
+  if [[ $ARG_ALL == 1 && ($ARG_MANAGER == 1 ||
+    $ARG_GC == 1 || $ARG_MONITOR == 1 || $ARG_COORDINATOR == 1 || $ARG_TSERVER == 1 ||
+    $ARG_SSERVER == 1 || $ARG_COMPACTOR == 1) ]]; then
+    echo -- "--all cannot be used with other options"
+    print_usage
+    exit 1
+  fi
+
+  # Handle the case where no args were passed
+  if [[ $ARG_MANAGER == 0 && $ARG_GC == 0 &&
+    $ARG_MONITOR == 0 && $ARG_COORDINATOR == 0 && $ARG_TSERVER == 0 &&
+    $ARG_SSERVER == 0 && $ARG_COMPACTOR == 0 ]]; then
+    ARG_ALL=1
+  fi
+
+  debug "ARG_ALL=$ARG_ALL"
+  debug "ARG_LOCAL=$ARG_LOCAL"
+  debug "ARG_MANAGER=$ARG_MANAGER"
+  debug "ARG_GC=$ARG_GC"
+  debug "ARG_MONITOR=$ARG_MONITOR"
+  debug "ARG_COORDINATOR=$ARG_COORDINATOR"
+  debug "ARG_TSERVER=$ARG_TSERVER"
+  debug "ARG_TSERVER_GROUP=$ARG_TSERVER_GROUP"
```

Comment:
`ARG_TSERVER_GROUP` does not apply in 2.1 and 3.1

Reply:
That's why it's never set when getopt options are parsed. It remains empty. This is just script internals to keep it aligned with what's in the main branch. It doesn't mean tserver groups are supported.

If you're looking at these changes by comparing with what's in 2.1, I suggest instead comparing it to what's in main. It's a much smaller diff, and makes a lot more sense.